src/main/java/pl/coderslab/web/AdminDeletePlan.java

```diff
@@ -0,0 +1,46 @@
+package pl.coderslab.web;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import pl.coderslab.dao.PlanDao;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+
+@WebServlet("/app/plan/delete")
+public class AdminDeletePlan extends HttpServlet {
+    public static final PlanDao planDao = new PlanDao();
+    public static final Logger logger = LogManager.getLogger(AdminDeletePlan.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        int planId = Integer.parseInt(req.getParameter("id"));
+        req.setAttribute("planId", planId);
+        getServletContext().getRequestDispatcher("/app-plan-delete-confirmation.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        HttpSession session = req.getSession();
+        int planId = Integer.parseInt(req.getParameter("id"));
+        String isPlanToDelete = req.getParameter("delete");
+
+        int adminId = (int) session.getAttribute("adminId");
+        logger.info("Plan id to delete {}", planId);
+
+        if (isPlanToDelete.equals("true")) {
+            try {
+                planDao.delete(planId);
```

Comment:
Usuwanie planu zadziała tylko wtedy kiedy nie są dodane żadne przepisy.
W przeciwnym razie, dostaniemy błąd sql odnośnie constraints z tabeli `recipe_plan`.
Więc pewnie trzeba będzie wyrzucić wszystkiego rekordy z tej tabeli z id planu, który zamierzamy usunać.

Reply:
Wrzucę query do usuwania przepisu z planu - to część zadania, które wziąłem.

Reply:
Faktycznie testowałem na planach bez przepisów :D 
Tak mi się wydaje ze tam powinno być on delete cascade - zeby rekordy z  recipe_plan się tez usunęły.  
Zaraz zerknę.

Reply:
no tak też myślę, że to lepsza opcja niż pisanie do tego 2 zapytań, bo ten sam problem będzie przy usuwaniu przepisu z listy przepisów użytkownika.

Reply:
z tego co wyczytałem, trzeba zrobić drop tabeli i ją utworzyć jeszcze raz, żeby dodać `ON DELETE CASCADE`

Reply:
Dokładnie - to logika podpowiada, żeby on cascade delete usuwać. Tak jak mowisz, z przepisem będzie tak samo - nie może istnieć recipe_plan rekord jeśli jego komponent został usunięty.